### DISCUSSION TITLE
Server listening to multiple ports?

### DISCUSSION THREAD

**Discussion by evesdropper** [grpc/grpc-java#11646] [Q&A] [answered]
Working on something where mulitple clients are sending grpc commands to a server, and I'm thinking the solution would be having the server listen to multiple ports and have each client connect through a different port.

I saw #5067 and it seems like grpc for Java supports multiple ports but I could not find any documentation anywhere; could someone point me to the right place or provide an example?

**Comment by ejona86** [comment, marked as answer]
Generally the approach is "start multiple Servers." There's pretty low cost to doing so.

To do was added in 5067, you'd use [NettyServerBuilder.addListenAddress()](https://grpc.github.io/grpc-java/javadoc/io/grpc/netty/NettyServerBuilder.html#addListenAddress(java.net.SocketAddress)). It has limitations like "not being able to change ports once started."
